lib/remoteproc/remoteproc.c

```diff
@@ -55,14 +55,14 @@ remoteproc_get_mem(struct remoteproc *rproc, const char *name,
 
 			pa_start = mem->pa;
 			pa_end = pa_start + mem->size;
-			if (pa >= pa_start && (pa + size) <= pa_end)
+			if (pa >= pa_start && (pa + size) < pa_end)
```

Comment:
(pa + size) = pa_end is a valid use case if size is not equal to 0.
That why i suggested you in https://github.com/OpenAMP/open-amp/issues/335 to return an error if pa >= pa_end

Reply:
Hi Arnopo,

Thank you for your explain.
The pa >= pa_end condition is really a best method.
I will modify it and send PR again.

Thanks,
Joshua